core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1521,8 +1521,15 @@ abstract class RDD[T: ClassTag](
     }
 
     checkpointData match {
-      case Some(reliable: ReliableRDDCheckpointData[_]) => logWarning(
-        "RDD was already marked for reliable checkpointing: overriding with local checkpoint.")
+      case Some(reliable: ReliableRDDCheckpointData[_]) =>
+        if (isCheckpointed) {
```

Comment:
can you rename this to `isCheckpointedAndMaterialized` and update the java docs? Right now it's a little confusing because it can be interpreted as "marked for checkpointing" only, which is actually what the existing java docs say.